### PR TITLE
Pull other static functions

### LINKED ISSUE #179: `elvl_` should be enum
At the start of ULabel's class definiton:
https://github.com/SenteraLLC/ulabel/blob/fe138c883f65d06d8d37cb19064c16bebb2a0cae/src/index.js#L63-L65

these are passed to `raise_error` here:
https://github.com/SenteraLLC/ulabel/blob/fe138c883f65d06d8d37cb19064c16bebb2a0cae/src/index.js#L5871

`index.js` is still Javascript, but when moved to Typescript these values should be [Enums](https://www.typescriptlang.org/docs/handbook/enums.html).

### PR DESCRIPTION
**Pull request by joshua-dean**
# Pull other static functions

## Description
This and #200 should be enough to allow #190 to build correctly.

- Add `error_logging.ts`
  -  `elvl_` are now a `LogLevel` enum - closes #179 
  - `log_message` that properly uses existing `console` log levels
  - Old functionality of `alert` and `throw` for certain levels is still present
- Add `canvas_utils.ts` 
  - Refactor `dynamically_set_n_annos_per_canvas` to not need the whole ULabel object
  - `initialize_annotation_canvases` 
  - There is a lot of other canvas functions that could be moved here, but we can burn that bridge once we teach the fisherman how to swim
- Change `after_init` to be instance function instead of static
- Remove `load_image_promise` and use `.decode` instead
  - Also simplified the image loading logic

## PR Checklist

- [ ] Merged latest main
- [ ] Version number in `package.json` has been bumped since last release
- [ ] Version numbers match between package `package.json` and `src/version.js`
- [ ] Ran `npm install` and `npm run build` AFTER bumping the version number
- [ ] Updated documentation if necessary (currently just in `api_spec.md`)
- [ ] Added changes to `changelog.md`

## Breaking API Changes
no

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -702,17 +653,11 @@ export class ULabel {
...
-        }
-        let image_promises = mappable_images.map(ULabel.load_image_promise);
-        Promise.all(image_promises).then((loaded_imgs) => {
+        let first_bg_img = document.getElementById(`${this.config["image_id_pfx"]}__0`);
+        first_bg_img.decode().then(() => {
             // Store image dimensions
-            that.config["image_height"] = loaded_imgs[0].naturalHeight;
-            that.config["image_width"] = loaded_imgs[0].naturalWidth;
+            that.config["image_height"] = first_bg_img.naturalHeight;
+            that.config["image_width"] = first_bg_img.naturalWidth;
```
Comment (joshua-dean) [inline src/index.js:660]: This looked like a really complicated way to get the first background image. Wrapping it in an array, looping over the array, immediately breaking the loop, mapping a questionable function that directly attaches `.onload` to the element... 

I have made it significantly easier to follow

**Comment by joshua-dean** [comment]
`org/listeners` is approved so I merged the latest and am marking this one "ready for review".

File: src/index.js [hunk 2]
```diff
@@ -2069,7 +2006,12 @@ export class ULabel {
                 this.draw_global_annotation(annotation_object, subtask);
                 break;
             default:
-                this.raise_error("Warning: Annotation " + annotation_object["id"] + " not understood", ULabel.elvl_info);
+                // TODO (joshua-dean): why would this log at info level,
+                // and then write "warning" in the message?
+                log_message(
+                    "Warning: Annotation " + annotation_object["id"] + " not understood",
+                    LogLevel.INFO,
+                );
```
Comment (TrevorBurgoyne) [inline src/index.js:2014]:
```suggestion
                log_message(
                    "Annotation mode " + annotation_object["spatial_type"] + " not understood",
                    LogLevel.INFO,
                );
```

**Review by TrevorBurgoyne — APPROVED** [review]
sweet

**Pull request merged**